src/test/java/org/opensearch/knn/plugin/transport/GetModelResponseTests.java

```diff
@@ -23,8 +27,22 @@
 
 import java.io.IOException;
 
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
 public class GetModelResponseTests extends KNNTestCase {
 
+    private static MockedStatic<KNNClusterUtil> knnClusterUtilMockedStatic;
+
+    @BeforeClass
+    public static void setup() {
```

Comment:
Could you tell why this is related with flakiness of tests?

Reply:
For some reason, the getClusterMinVersion() method in KNNClusterUtil returns a random version, not the most recent.  So the unit test would fail when the random version was before 2.12 or 3.0, since the field added wasn't in those.  This change ensures that the current version for each branch is returned.

Reply:
The randomness of the version might be an intended behavior. I am afraid your code here change that behavior for other test cases as well.

We need to check if there are any other tests which rely on randomness of the version. Also, changing behavior of static method should be done for entire test level but not in individual test level(Maybe inside KNNTestCase class?) so that there won't be any race condition among tests.

Reply:
Wouldn't the variable be localized to the class, unless it was specifically accessed elsewhere?

I was trying to only make this change for the classes that required a version check, so that any intended use for the randomized version would stay the same for the rest of the tests.

Reply:
You are using static mock which won't be scoped to your test automatically.
https://www.baeldung.com/mockito-mock-static-methods

Reply:
I'll attempt to use the try method suggested in that article